### DISCUSSION TITLE
Import Reveal.js

### DISCUSSION THREAD

**Discussion by Helder-SR** [vercel/next.js#43631] [Help] [answered]
When I'm trying to import Reveal.js on my app, I have a problem: navigator is not defined.
It's just this:
`import Reveal from 'reveal.js';`
`import Markdown from 'reveal.js/plugin/markdown/markdown.esm.js';`

`let deck = new Reveal({`
`   plugins: [ Markdown ]`
`})`
`deck.initialize();`

**Comment by icyJoseph** [comment, marked as answer]
The problem here is that during SSR there's no window/dom or anything a browser environment provides.

You have to code in such a way that this code is executed only on the client.

You can do that with `next/dynamic`, with `ssr: false`: https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr

Let me know if that's enough.

**Reply by Helder-SR** [reply]
I'm not sure if I understood correctly, I'm new to next.js, but if you do:
`const Reveal = dynamic(() => import('reveal.js'), {
  ssr: false,
})`
"Reveal" wouldn't be a constructor

**Reply by icyJoseph** [reply]
Right, that's not what I am proposing. Let's say you create:

```js
// my-module.js
import Reveal from 'reveal.js';
import Markdown from 'reveal.js/plugin/markdown/markdown.esm.js';

let deck = new Reveal({
plugins: [ Markdown ]
})
deck.initialize();

const MyComponent = () => {
  // do something with deck 
}

export default MyComponent
```

Then when you need to use this component which uses Reveal:

```
const Reveal = dynamic(() => import('./my-component'), { ssr: false, })
```

## Alternative

You could also try to load Reveal as a dynamic import

```jsx
const Loader = () => {
    const [mod, setMod] = useState(null)
    useEffect(() => { 
            import("reveal.js").then(setMod)
      , [])

    if(!mod) return <div>Loading...</div>

   return <ComponentThatusesReveal reveal={reveal} />
}
```

You could also do all of the Reveal setup in a module and load it via dynamic import (not next/dynamic)

**Reply by Helder-SR** [reply]
Thank you I finally understood !
